### PR TITLE
Phishing Dedup enhancements 

### PR DESCRIPTION
**Pull request by erezh31**
## Status
- [ ] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Description
Add arguments - close as duplicate & set threshold
change result view human readable

## Screenshots
The view of the incident results:
image

### CODE REVIEW

File: Packs/Phishing/Scripts/PhishingDedupPreprocessingRule/PhishingDedupPreprocessingRule.py [hunk 1]
```diff
@@ -231,14 +231,15 @@ def return_entry(message, existing_incident=None):
...
+    if demisto.args().get('closeAsDuplicate', 'true') == 'true':
+        res = demisto.executeCommand("CloseInvestigationAsDuplicate", {
+            'duplicateId': existing_incident['id']})
+
+        if is_error(res):
+            return_error(res)
+    formatted_incident = format_similar_incident(existing_incident, similarity)
+    message = tableToMarkdown("Duplicate incident found with similarity {:.1f}%".format(similarity * 100),
+                              formatted_incident)
     message += 'This incident will be closed and linked to #{}.'.format(existing_incident['id'])
```
Comment (lperryp) [inline Packs/Phishing/Scripts/PhishingDedupPreprocessingRule/PhishingDedupPreprocessingRule.py:243]: change the message based on `demisto.args().get('closeAsDuplicate', 'true') == 'true'`

File: Packs/Phishing/Scripts/PhishingDedupPreprocessingRule/PhishingDedupPreprocessingRule.py [hunk 2]
```diff
@@ -247,10 +248,20 @@ def create_new_incident():
...
+            'Time': str(incident['created']),
+            'Email from': incident.get(demisto.args().get('emailFrom')),
+            'Text Similarity': "{:.1f}%".format(similairy * 100),
+            }
+
+
+def create_new_incident_low_similarity(existing_incident, new_incident, similarity):
+    message = '# This incident is not a duplicate of an existing incident.\n'
+    formatted_incident = format_similar_incident(existing_incident, similarity)
+    message += tableToMarkdown("Most similar incident found", formatted_incident)
```
Comment (lperryp) [inline Packs/Phishing/Scripts/PhishingDedupPreprocessingRule/PhishingDedupPreprocessingRule.py:264]: perhaps we should add a minimum threshold of displaying the original message, or even 2 different messages for a certain threshold
meaning:  if x<similarity<0.99 - show most similar message
if similarity < x :  we can just say that no duplicated incident was found
for x=0.8 or x=0.85
what do you think?

File: Packs/Phishing/Scripts/PhishingDedupPreprocessingRule/PhishingDedupPreprocessingRule.py [hunk 3]
```diff
@@ -247,10 +248,20 @@ def create_new_incident():
...
+    return {'Id': "[%s](#/Details/%s)" % (incident['id'], incident['id']),
+            'Name': incident['name'],
+            'Closed': incident.get('closed') != "0001-01-01T00:00:00Z",
+            'Time': str(incident['created']),
+            'Email from': incident.get(demisto.args().get('emailFrom')),
+            'Text Similarity': "{:.1f}%".format(similairy * 100),
+            }
+
+
+def create_new_incident_low_similarity(existing_incident, new_incident, similarity):
```
Comment (lperryp) [inline Packs/Phishing/Scripts/PhishingDedupPreprocessingRule/PhishingDedupPreprocessingRule.py:261]: not critical but seems that `new_incident` is not required for this func

File: Packs/Phishing/Scripts/PhishingDedupPreprocessingRule/PhishingDedupPreprocessingRule.py [hunk 4]
```diff
@@ -247,10 +248,20 @@ def create_new_incident():
...
 
-def create_new_incident_low_similarity(existing_incident, similarity):
-    message = 'This incident is not a duplicate of an existing incident.\n'
-    message += 'Most similar incident found is #{} with similarity of {:.1f}%.\n'.format(existing_incident['id'],
-                                                                                         similarity * 100)
+def format_similar_incident(incident, similairy):
+    return {'Id': "[%s](#/Details/%s)" % (incident['id'], incident['id']),
+            'Name': incident['name'],
+            'Closed': incident.get('closed') != "0001-01-01T00:00:00Z",
+            'Time': str(incident['created']),
```
Comment (lperryp) [inline Packs/Phishing/Scripts/PhishingDedupPreprocessingRule/PhishingDedupPreprocessingRule.py:255]: perhaps adding more friendly formatting for `created` timedate

**Review by lperryp — CHANGES_REQUESTED** [review]
change is great and important small comments s

File: Packs/Phishing/Scripts/PhishingDedupPreprocessingRule/PhishingDedupPreprocessingRule.py (hunk 4 above)
Comment (erezh31) [reply Packs/Phishing/Scripts/PhishingDedupPreprocessingRule/PhishingDedupPreprocessingRule.py:255]: it's actually looks fine it's timestamp formatting

File: Packs/Phishing/Scripts/PhishingDedupPreprocessingRule/PhishingDedupPreprocessingRule.py (hunk 2 above)
Comment (erezh31) [reply Packs/Phishing/Scripts/PhishingDedupPreprocessingRule/PhishingDedupPreprocessingRule.py:264]: agree

File: Packs/Phishing/Scripts/PhishingDedupPreprocessingRule/PhishingDedupPreprocessingRule.py [hunk 5]
```diff
@@ -258,15 +258,17 @@ def format_similar_incident(incident, similairy):
...
-    return_entry(message)
+
+    if similarity > SIMILARITY_THRESHOLD - CLOSE_TO_SIMILAR_DISTANCE:
+        formatted_incident = format_similar_incident(existing_incident, similarity)
+        message += tableToMarkdown("Most similar incident found", formatted_incident)
+        message += 'The threshold for considering 2 incidents as duplicate is a similarity ' \
+                   'of {:.1f}%.\n'.format(SIMILARITY_THRESHOLD * 100)
+        message += 'Therefore these 2 incidents will not be considered as duplicate and the current incident ' \
+                   'will remain active.\n'
+        return_entry(message)
```
Comment (lperryp) [inline Packs/Phishing/Scripts/PhishingDedupPreprocessingRule/PhishingDedupPreprocessingRule.py:272]: I think it's also good to return a message  for "else" case which can be the same message as other cases when no incident found

**Review by lperryp — APPROVED** [review]

File: Packs/Phishing/Scripts/PhishingDedupPreprocessingRule/PhishingDedupPreprocessingRule.py (hunk 5 above)
Comment (erezh31) [reply Packs/Phishing/Scripts/PhishingDedupPreprocessingRule/PhishingDedupPreprocessingRule.py:272]: the first line said that no duplicate incident found

Comment (lperryp) [reply Packs/Phishing/Scripts/PhishingDedupPreprocessingRule/PhishingDedupPreprocessingRule.py:272]: the return is within the if, don't think you return it

Comment (erezh31) [reply Packs/Phishing/Scripts/PhishingDedupPreprocessingRule/PhishingDedupPreprocessingRule.py:272]: good point, fixed

**Comment by erezh31** [comment]
@kirbles19 small changes - RN & add argument threshold - copy it's description from other automation we have with similar meaning

File: Packs/Phishing/Scripts/PhishingDedupPreprocessingRule/PhishingDedupPreprocessingRule.py [hunk 6]
```diff
@@ -290,12 +292,12 @@ def main():
     EMAIL_HTML_FIELD = input_args.get('emailBodyHTML', EMAIL_HTML_FIELD)
     FROM_FIELD = input_args.get('emailFrom', FROM_FIELD)
     FROM_POLICY = input_args.get('fromPolicy', FROM_POLICY)
-    existing_incidents = get_existing_incidents(input_args)
+    new_incident = demisto.incidents()[0]
+    existing_incidents = get_existing_incidents(input_args, new_incident.get('type'))
```
Comment (lperryp) [inline Packs/Phishing/Scripts/PhishingDedupPreprocessingRule/PhishingDedupPreprocessingRule.py:296]: new_incident.get('type', None)

**Pull request merged**